### ISSUE TITLE
futhark-test should report the actual number of tests run

### ISSUE THREAD

**Issue by lolkat2k** [diku-dk/futhark#421]
When running `futhark-test` on the following program:
```
-- Test the add function.
-- ==
-- entry: add
-- input { 1 } output { 2 }
-- input { 2 } output { 3 }

entry add (x : i32) : i32 = x + 1
```
it outputs `(0 failed, 1 passed)`.

This is due to `futhark-test` counting the number of files and not the number of tests run. If we take an input-output pair as being one test, then the output should be `(0 failed, 2 passed)`.

**Comment by lolkat2k** [comment]
Note that, if `futhark-test` is given a directory with x `.fut`-files, `futhark-test` will output `(0 failed, x)` passed.

**Comment by Athas** [comment]
This would be relatively straightforward to fix, I think.  `futhark-test` uses files as the basic unit of labour (which should not change because that's also the unit of compilation), but each file could easily count as several tests.  Or maybe a separate statistic could count the number of test cases run.  All files have their test block parsed before anything starts executing, so we have access to all the information we need.

**Comment by zfnmxt** [comment]
Fixed by 8d88228.
